sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1992,6 +1992,30 @@ object GpuOverrides {
         override def convertToGpu(): GpuExpression =
           GpuCreateNamedStruct(childExprs.map(_.convertToGpu()))
       }),
+    expr[ArrayContains](
+      "Returns a boolean if the array contains the passed in key",
+      ExprChecks.binaryProjectNotLambda(
+        TypeSig.BOOLEAN,
+        TypeSig.BOOLEAN,
+        ("array", TypeSig.ARRAY.nested(TypeSig.commonCudfTypes + TypeSig.NULL),
+          TypeSig.ARRAY.nested(TypeSig.all)),
+        ("key", TypeSig.commonCudfTypes, TypeSig.all)),
+      (in, conf, p, r) => new BinaryExprMeta[ArrayContains](in, conf, p, r) {
+        override def tagExprForGpu(): Unit = {
+          if (extractLit(in.left).isDefined) {
+            willNotWorkOnGpu("Literal arrays are not supported for array_contains")
+          }
+          if (conf.hasNans && (in.children.map(a => a.dataType).contains(DoubleType) ||
+            in.children.map(a => a.dataType).contains(FloatType))) {
+            willNotWorkOnGpu("Array Contains on floating point columns that can contain NaNs " +
```

Comment:
Just curious is this true even if what we are looking for is a literal that is not NaN?  That feels like a common case and we might be able to allow it to work.

Reply:
Sorry for the delay, I am running some tests and fighting Spark a bit with the plans with literals but I do have a fix. Will post asap.

Reply:
As an update , if we look at : 
```
array_contains(col('a'), col('a')[5]))
```
in the pytests, does not exercise the scalar for rhs route, I am going to work on getting a scalar that actually exists in the lists.

Reply:
I have added a test to check for a literal from the array that is not a nan to see if we get the right results. It needed a `collect()` and I appreciate any improvements if needed.